lib/buffer.js

```diff
@@ -1021,557 +1002,764 @@ Buffer.prototype.slice = function slice(start, end) {
   }
+  throw new errors.RangeError('ERR_OUT_OF_RANGE',
+                              'byteLength',
+                              '>= 1 and <= 6');
 }
 
+function checkInt(value, offset, min, max) {
+  checkOffsetType(offset);
+  if (value > max || value < min) {
+    throw new errors.RangeError('ERR_INVALID_OPT_VALUE', 'value', value,
+                                `Range: value <= ${max} && value >= ${min}`);
+  }
+}
 
-Buffer.prototype.readUIntLE =
-  function readUIntLE(offset, byteLength, noAssert) {
-    offset = offset >>> 0;
-    byteLength = byteLength >>> 0;
-    if (!noAssert) {
-      checkByteLength(byteLength);
-      checkOffset(offset, byteLength, this.length);
-    }
+Buffer.prototype.readUIntLE = function readUIntLE(offset, byteLength) {
+  if (byteLength === 6)
+    return readUInt48LE(this, offset);
+  if (byteLength === 5)
+    return readUInt40LE(this, offset);
+  if (byteLength === 3)
+    return readUInt24LE(this, offset);
+  if (byteLength === 4)
+    return this.readUInt32LE(offset);
+  if (byteLength === 2)
+    return this.readUInt16LE(offset);
+  if (byteLength === 1)
+    return this.readUInt8(offset);
+
+  byteLengthError(byteLength);
+};
 
-    var val = this[offset];
-    var mul = 1;
-    var i = 0;
-    while (++i < byteLength && (mul *= 0x100))
-      val += this[offset + i] * mul;
+function readUInt48LE(buf, offset) {
+  checkOffsetType(offset);
+  const first = buf[offset];
+  const last = buf[offset + 5];
+  if (first === undefined || last === undefined)
+    boundsError(offset, 6, buf.length);
+
+  return (buf[offset + 4] | (last << 8)) * SHIFT_LEFT_32 +
+    (first |
+      (buf[offset + 1] << 8) |
+      (buf[offset + 2] << 16)) +
+    (buf[offset + 3] * 0x1000000);
+}
 
-    return val;
-  };
+function readUInt40LE(buf, offset) {
+  checkOffsetType(offset);
+  const first = buf[offset];
+  const last = buf[offset + 4];
+  if (first === undefined || last === undefined)
+    boundsError(offset, 4, buf.length);
+
+  return last * SHIFT_LEFT_32 +
+    (first |
+      (buf[offset + 1] << 8) |
+      (buf[offset + 2] << 16)) +
+    (buf[offset + 3] * 0x1000000);
+}
 
+function readUInt24LE(buf, offset) {
+  checkOffsetType(offset);
+  const first = buf[offset];
+  const last = buf[offset + 2];
+  if (first === undefined || last === undefined)
+    boundsError(offset, 3, buf.length);
 
-Buffer.prototype.readUIntBE =
-  function readUIntBE(offset, byteLength, noAssert) {
-    offset = offset >>> 0;
-    byteLength = byteLength >>> 0;
-    if (!noAssert) {
-      checkByteLength(byteLength);
-      checkOffset(offset, byteLength, this.length);
-    }
+  return (last << 16) |
+    (first | (buf[offset + 1] << 8));
+}
 
-    var val = this[offset + --byteLength];
-    var mul = 1;
-    while (byteLength > 0 && (mul *= 0x100))
-      val += this[offset + --byteLength] * mul;
+Buffer.prototype.readUIntBE = function readUIntBE(offset, byteLength) {
+  if (byteLength === 6)
```

Comment:
Same as above, re: switch. More of these down below, won't mention every single one :)